### PR TITLE
Football club wikipedia scraper for issue #466

### LINKED ISSUE #466: Script to get a Football Player's Club History from Wikipedia
# Description

A Script to get the Club History (with noteworthy years, if any are specified in Wikipedia) of a Football Player by scraping that footballer's Wikipedia Page.

## Language

 [✔] **Python**

## Type of issue

[✔] Web-Scraping Script

### PR DESCRIPTION
**Pull request by RohiniRG**
# Description

Script which scrapes Wikipedia to obtain Football players' club careers over the years, using Python's BeautifulSoup library.

Fixes #466 

## Type of change

_Please delete options that are not relevant._

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] Documentation Update

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

File: Python/Wikipedia_Scraper/main.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+from bs4 import BeautifulSoup
+import requests
+
+
+def get_player_name():
+    """ Takes Football player's name input from the user and
+    returns a string suitable to concatenate in the wikipedia URL. """
```
Comment (vybhav72954) [inline Python/Wikipedia_Scraper/main.py:7]: There needs to be no whitespace after the triple quotes for string, alternatively what you should prefer is a white line, what I basically mean is this - 
BAD
```
    """ Takes Football player's name input from the user and
    returns a string suitable to concatenate in the wikipedia URL. """
```

GOOD

```
    """Takes Football player's name input from the user and
    returns a string suitable to concatenate in the wikipedia URL."""
```

BEST

```
"""
Takes Football player's name input from the user and
returns a string suitable to concatenate in the wikipedia URL.
"""
```

File: Python/Wikipedia_Scraper/main.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    for i in player:
+        name = i.capitalize()
+        player_name += name
+        player_name += ' '
+
+    return player_name
+
+
+def get_club_info(headers):
+    """ Stores only the information regarding the club career of
```
Comment (vybhav72954) [inline Python/Wikipedia_Scraper/main.py:20]: Change the docstring similarly.

File: Python/Wikipedia_Scraper/main.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        return club_data
+
+    # In case the name given is not a Football player
+    except TypeError:
+        print('INVALID PLAYER NAME!!')
+        return 0
+
+
+def display_info(club_data):
+    """ To display the club career information of player collected. """
```
Comment (vybhav72954) [inline Python/Wikipedia_Scraper/main.py:47]: Change the docstring similarly.

File: Python/Wikipedia_Scraper/main.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    for item in club_data:
+        if item[:4].isdigit():
+            print(f'    {item}')
+        else:
+            print()
+            print(f'CLUB NAME: {item}')
+
+
+def main():
+    print()
```
Comment (vybhav72954) [inline Python/Wikipedia_Scraper/main.py:58]: A blank print statement?
If you want a new line use escape character.

Else consider removing it!

File: Python/Wikipedia_Scraper/main.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+    for item in club_data:
+        if item[:4].isdigit():
+            print(f'    {item}')
+        else:
+            print()
+            print(f'CLUB NAME: {item}')
+
+
+def main():
```
Comment (vybhav72954) [inline Python/Wikipedia_Scraper/main.py:57]: A docstring here is also recommended.

File: Python/Wikipedia_Scraper/main.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    print()
+    print('WELCOME!!!HERE YOU CAN GET THE CLUB CAREER INFORMATION'
+          ' REGARDING YOUR FAVOURITE FOOTBALL PLAYER FROM WIKIPEDIA!')
+    while True:
+        print()
```
Comment (vybhav72954) [inline Python/Wikipedia_Scraper/main.py:62]: Blank print statement?

File: Python/Wikipedia_Scraper/README.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+- Using BeautifulSoup, a python library useful for web scraping, this script helps to scrape Wikipedia to obtain a Football player's club career data
+
+- In this script, we take the input of the name of any popular Football player from the user and generate the club career details associated with them.
+
+## Setup Instructions
+
+### Requirements
+
+1. BeautifulSoup library:
```
Comment (vybhav72954) [inline Python/Wikipedia_Scraper/README.md:11]: Considering you have a `requirements.txt`

Consider using `pip3 install -r requirements.txt`

in place of bs4 and request

You can list out dependencies if you want as list though

**Review by vybhav72954 — CHANGES_REQUESTED** [review]
I have requested a few minor changes

**Comment by RohiniRG** [comment]
@vybhav72954 I made the changes you mentioned, please have a look. Thank you!

**Comment by vybhav72954** [comment]
@iamakkkhil Why do i see a commit from you in the working tree?
@RohiniRG ?

**Comment by iamakkkhil** [comment]
I don't know what is happening. This is the very first time we are submitting the PR. 
Can you let us know what to do now @vybhav72954 ?
Apologies for any inconvenience.

**Comment by RohiniRG** [comment]
> @iamakkkhil Why do i see a commit from you in the working tree?
> @RohiniRG ?

Actually @iamakkkhil had the last merge in this repo and when I tried to push this commit, his previously merged files also got committed I guess... I am not really sure how this happened. What have I done wrong and how can I rectify this?

**Comment by RohiniRG** [comment]
Sorry for this mess! I am just trying to figure out git works.

**Comment by HarshCasper** [comment]
No worries @RohiniRG 

Would you be working on the same PR or make a new branch and raise a new PR? Do let us know about it. 

P.S: We want you to make mistakes because failing is sometimes a better way to learn something. Not success :)

**Comment by RohiniRG** [comment]
> No worries @RohiniRG
> 
> Would you be working on the same PR or make a new branch and raise a new PR? Do let us know about it.
> 
> P.S: We want you to make mistakes because failing is sometimes a better way to learn something. Not success :)

Thank you! I would like to make a new branch and start with a new PR again.

**Pull request closed without merging**